File: metronic_asp.net-core_v8.2.3/asp.net-core/metronic_asp.net-core_v8.2.3/metronic_asp.net-core_v8.2.3/Starterkit/_keenthemes/libs/KTThemeSettings.cs
namespace Starterkit._keenthemes.libs;

class KTThemeSettings
{
    public static KTThemeBase Config { get; set; } = new KTThemeBase();

    public static void init(IConfiguration configuration)
    {
        Config = configuration.GetSection("Theme").Get<KTThemeBase>() ?? Config;
    }
}

File: metronic_spring_v8.2.3/spring/metronic_spring_v8.2.3_demo1/spring_demo1/starterkit/src/main/java/com/theme/starterkit/libs/config/KTThemeSettings.java
package com.theme.starterkit.libs.config;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.File;

public class KTThemeSettings {
    public KTThemeBaseConfig config;

    public KTThemeSettings() {
        ObjectMapper objectMapper = new ObjectMapper();
        File file = new File("_keenthemes/config/themesettings.json");
        try {
            config = objectMapper.readValue(file, KTThemeBaseConfig.class);
        } catch (Exception e) {
            throw new IllegalStateException(e);
        }
    }
}
